src/extension/background-script/connectors/galoy.ts

```diff
@@ -108,36 +110,157 @@ class Galoy implements Connector {
       },
     };
 
-    const { tokens: amountInSats, payment: paymentHash } = parsePaymentRequest({
+    const { tokens: amountInSats, id: paymentHash } = parsePaymentRequest({
       request: args.paymentRequest,
     });
 
     return this.request(query).then((data) => {
+      if (data.error || data.errors) {
+        const error = data.error || data.errors;
+        const errMessage = error?.errors?.[0]?.message || error?.[0]?.message;
+        return {
+          error: errMessage || JSON.stringify(error),
+        };
+      }
       if (data.data.lnInvoicePaymentSend.errors?.[0]?.message) {
         return { error: data.data.lnInvoicePaymentSend.errors[0].message };
       }
-      if (data.error) {
-        return {
-          error: data.error?.errors?.[0]?.message || JSON.stringify(data.error),
-        };
+      switch (data.data.lnInvoicePaymentSend.status) {
+        case "ALREADY_PAID":
+          return {
+            error: "Invoice was already paid.",
+          };
+        case "FAILURE":
+          return {
+            error: "Payment failed.",
+          };
+        case "PENDING":
+          return {
+            data: {
+              preimage: "No preimage, payment still pending",
+              paymentHash,
+              route: { total_amt: amountInSats, total_fees: 0 },
+            },
+          };
+        default:
+          return {
+            data: {
+              preimage: "No preimage received",
+              paymentHash,
+              route: { total_amt: amountInSats, total_fees: 0 },
+            },
+          };
       }
-      return {
-        data: {
-          preimage: "No preimage received",
-          paymentHash,
-          route: { total_amt: amountInSats, total_fees: 0 },
-        },
-      };
     });
   }
 
   async checkPayment(args: CheckPaymentArgs): Promise<CheckPaymentResponse> {
-    return {
-      data: {
-        paid: false,
-        preimage: "",
-      },
+    const TRANSACTIONS_PER_PAGE = 20;
+
+    const query = {
+      query: `
```

Comment:
We probably need an easier way to query single transactions than this 🤕  (cc @samerbuna @dolcalmi @daviroo)